src/osd/OSD.h

```diff
@@ -1544,7 +1544,7 @@ class OSD : public Dispatcher,
   void register_session_waiting_on_pg(Session *session, spg_t pgid) {
     Mutex::Locker l(session_waiting_lock);
     set<Session*> &s = session_waiting_for_pg[pgid];
-    set<Session*>::iterator i = s.find(session);
+    set<Session*>::const_iterator i = s.find(session);
```

Comment:
i'd not suggest use `set<Session*>::const_iterator` here, unless we have a simple way to avoid the unnecessary copy ctor.